lambdas/check-gpg45-score/src/main/java/uk/gov/di/ipv/core/checkgpg45score/CheckGpg45ScoreHandler.java

```diff
@@ -0,0 +1,163 @@
+                                .with(LOG_SCORE_TYPE.getFieldName(), scoreType));
+                return new JourneyResponse(JOURNEY_UNMET_PATH).toObjectMap();
+            }
+        } catch (HttpResponseExceptionWithErrorBody e) {
+            LOGGER.error("Received HTTP response exception", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH, e.getResponseCode(), e.getErrorResponse())
+                    .toObjectMap();
+        } catch (ParseException e) {
+            LOGGER.error("Unable to parse GPG45 scores from existing credentials", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                            ErrorResponse.FAILED_TO_PARSE_ISSUED_CREDENTIALS)
+                    .toObjectMap();
+        } catch (UnknownEvidenceTypeException e) {
+            LOGGER.error("Unable to determine type of credential", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                            ErrorResponse.FAILED_TO_DETERMINE_CREDENTIAL_TYPE)
+                    .toObjectMap();
+        } catch (UnknownScoreTypeException e) {
+            LOGGER.error("Unable to process score type", e);
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                            ErrorResponse.UNKNOWN_SCORE_TYPE)
+                    .toObjectMap();
+        }
+    }
+
+    private List<SignedJWT> getParsedCredentials(String ipvSessionId) throws ParseException {
+        IpvSessionItem ipvSessionItem = ipvSessionService.getIpvSession(ipvSessionId);
+        ClientOAuthSessionItem clientOAuthSessionItem =
+                clientOAuthSessionDetailsService.getClientOAuthSession(
+                        ipvSessionItem.getClientOAuthSessionId());
+        String userId = clientOAuthSessionItem.getUserId();
+
+        String govukSigninJourneyId = clientOAuthSessionItem.getGovukSigninJourneyId();
+        LogHelper.attachGovukSigninJourneyIdToLogs(govukSigninJourneyId);
+
+        return gpg45ProfileEvaluator.parseCredentials(
+                userIdentityService.getUserIssuedCredentials(userId));
+    }
+
+    private int getScore(String ipvSessionId, String scoreType)
+            throws ParseException, UnknownEvidenceTypeException, UnknownScoreTypeException {
+        List<SignedJWT> credentials = getParsedCredentials(ipvSessionId);
+        Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(credentials);
+        return switch (scoreType) {
+            case FRAUD -> gpg45Scores.getFraud();
+            case ACTIVITY -> gpg45Scores.getActivity();
+            case VERIFICATION -> gpg45Scores.getVerification();
```

Comment:
We don't have anything here for evidence. But it's a little trickier to implement due to the nature of it - having two elements and potentially have multiple values. I think it's ok to leave out for now and add in if/when we need it.